docs/README.rst

```diff
@@ -20,40 +20,107 @@ Documentation
+| My edits contain tables or           | Local Editor     | GitHub can provide Markdown previews, but might |
+| other formatting changes.            |                  | change ``.rst``` styling. Use a local build.    |
++--------------------------------------+------------------+-------------------------------------------------+
+| I want to make a new page/           | Local Editor     | Will need a local build to check navigation and |
+| delete a page.                       |                  | link redirects.                                 |
++--------------------------------------+------------------+-------------------------------------------------+
+| I want to edit autogenerated content | Either, probably | Allows you to easily find the correct file and  |
+| on a page.                           | Local Editor     | generate a preview before creating the PR.      |
++--------------------------------------+------------------+-------------------------------------------------+
+
+Finding source content to edit
+------------------------------
+
+Sphinx has _roles_ and _directives_, where Markdown docs frameworks often do not have similar functionality. This means that Airflow uses directives
+to pull code examples, autogenerate indexes and tables of contents, and reference from resources across the codebase and across documentation provider packages.
+This can make it confusing to find the content source on certain types of reference pages.
+
+For example, in `Command Line Interface and Environment Variables Reference <https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#environment-variables>`__, the CLI reference is `autogenerated <https://github.com/apache/airflow/blob/main/docs/apache-airflow/cli-and-env-variables-ref.rst?plain=1#L44>`__,
+and requires more complex scripting. While the `Environment Variables <https://github.com/apache/airflow/blob/main/docs/apache-airflow/cli-and-env-variables-ref.rst?plain=1#L51>`__ are explicitly written.
+
+To make an edit to an autogenerated doc, you need to make changes to a string in the Python source file. In the previous example, to make edits to the CLI command reference text, you must edit the `cli_config.py <https://github.com/apache/airflow/blob/main/airflow/cli/cli_config.py#L1861>`__ source file.
+
 Building documentation
 ======================
 
-To generate a local version you can use `<../BREEZE.rst>`_.
+To generate a local version of the docs you can use `<../BREEZE.rst>`_.
 
 The documentation build consists of verifying consistency of documentation and two steps:
 
 * spell checking
 * building documentation
 
-You can only run one of the steps via ``--spellcheck-only`` or ``--docs-only``.
+You can choose to run the complete build, to build all the docs and run spellcheck. Or, you can use the optional flags, ``--spellcheck-only``, ``--docs-only``, or ``--package-filter`` to choose the scope of the build.
```

Comment:
In the new version of `build-docs` (since ~ 3 months) you can simply specify package names as arguments without --package-filter - they are also autocomplete'able.